explainer.md

```diff
@@ -57,6 +57,25 @@ This explainer does not address Anchors (see [anchors](https://github.com/immers
 
 This API should be asynchronous in order to avoid main thread blocking calls. This also allows more flexibility in the underlying implementation. It is important to note that, due to the asynchronous nature of the API, hit results will always have at least 1 frame of latency.
 
+The requirement that the hit-test API is async creates some challenges. It has some ergonomics issues in the code style that arises from the promise-based call structure, but it also has a more critical flaw in the way the time discrepancy between the request and the return affect different use-cases. Currently, since we process a 3D-ray-based hit request based on the state of the world for the frame following the request, there is noticeable drift when using hit-test to create a reticle in 3D space. This drift comes from the frame delay between when the ray is generated by projecting into the world from screen space and when that ray is used to generate the hit for the reticle - the orientation of the device no longer matches the ray that was generated when the promise resolves.
+
+The following diagram should help in understanding the underlying issue:
+<img alt="Hit-test example" src="explainer-media/timing-diagram.png" style="float: left; object-fit: cover; width: 45%; height: 20em; margin-right: 1em; ">
+
+Situation: call requestHitTest(input ray) at frame N. 
+Problem: depending on the use-case, location 1 or location 2 may be the desired location of the result. Because the surface hit may be several meters away, small device rotations can result in large discrepancies between locations 1 and 2.
+
+Use-Cases:
+*   Reticle
+    *   For a pixel-perfect reticle, location 2 is the correct result. We need the reticle to be rendered in frame N+1 at location 2, otherwise it will appear to lag behind the device.
+    *   This means we either need to use a 2d API for reticles since 2d remains accurate from frame to frame (is there a way this could work for HMDs?), or we need to automatically correct the 3d input ray to the current device ray by adding the delta between the device transform at frame N and N+1.
```

Comment:
This doesn't seem to be 2D API specific; a vector (<0,0,1>) coming out from a mobile device should be the same as a vector coming out of an HMD, or an input device (if the vector can be resolved that is, as it could point out of screen). This sentence seems like it should be updated within the context of the last paragraph here.

Reply:
I was just mentioning that a 2D version of the API would not have this problem. If you pass in a 2d (0.5, 0.5) input, that inherently uses the most up-to-date device pose when it gets projected into the world.